apps/omg_db/lib/db.ex

```diff
@@ -31,41 +31,37 @@ defmodule OMG.DB do
   @callback blocks(block_to_fetch :: list()) :: {:ok, list(term)}
   @callback utxos() :: {:ok, list({utxo_pos_db_t, term})}
   @callback utxo(utxo_pos_db_t) :: {:ok, term} | :not_found
-  @callback exit_infos() :: {:ok, list(term)}
-  @callback in_flight_exits_info() :: {:ok, list(term)}
   @callback competitors_info() :: {:ok, list(term)}
-  @callback exit_info({pos_integer, non_neg_integer, non_neg_integer}) :: {:ok, map} | :not_found
   @callback spent_blknum(utxo_pos_db_t()) :: {:ok, pos_integer} | :not_found
   @callback block_hashes(integer()) :: {:ok, list()}
   @callback child_top_block_number() :: {:ok, non_neg_integer()} | :not_found
+  @callback get_single_value(atom()) :: {:ok, term} | :not_found
```

Comment:
So current DB is used to save data in two ways:
1. single variable: [list of them](https://github.com/omisego/elixir-omg/blob/e8d486d2cd19808d8a5e56d4a07412645813441c/apps/omg_db/lib/db.ex#L128-L150)
1. table-ish data, prefixed with some string before the "key" of accessing the data: [prefixes](https://github.com/omisego/elixir-omg/blob/e8d486d2cd19808d8a5e56d4a07412645813441c/apps/omg_db/lib/omg_db/rocksdb/core.ex#L28-L45) 

`get_single_value/1` is an already used function. This PR adds that here simply due to I think it was missed to list as one of the behavior. This function is currently used to get data for single variable storage.

This PR introduces: `get/2` and `get_all_by_type/1` which are used to get the table-ish, prefixed data.

Reply:
hmm...just realized today that we also need sth like `get_multi(type, list_of_keys)`. Perhaps remove `get/2` and rely on `get_multi/2` 🤔

Reason we need such `get_multi/2` is that we are going to read state from DB instead of process state. We need sth efficient to retrieve the needed data.